{ 
  "text" : "An issue was discovered in GitLab Community and Enterprise Edition through 12.2.1. Under certain circumstances, CI pipelines could potentially be used in a denial of service attack.\n" ,
  "entities" : [ { 
    "eid" : "E1" ,
    "keyid" : "ba83b651c775dd8a2b551f7f8b58a11d" ,
    "type" : "MiscEntity" ,
    "subType" : "pipeline" ,
    "value" : "CI pipeline" ,
    "begin" : 112 ,
    "end" : 124 ,
    "canonicalForm" : "CI pipeline" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E2" ,
    "keyid" : "Campaign#10e60b758ba8d4e7d9d47c2697d42f7b" ,
    "type" : "Campaign" ,
    "subType" : "Campaign" ,
    "value" : "denial of service attack" ,
    "begin" : 156 ,
    "end" : 180 ,
    "canonicalForm" : "denial of service attack" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E3" ,
    "keyid" : "TTP#9c98ef6a7451d6f82ca2194ee23d1acb" ,
    "type" : "TTP" ,
    "subType" : "TTP" ,
    "value" : "denial of service" ,
    "begin" : 156 ,
    "end" : 173 ,
    "canonicalForm" : "denial of service" ,
    "score" : "0.0"
  }
  ,  ] ,
  "relations" : [ { 
    "rid" : "R1" ,
    "type" : "used_in" ,
    "predicate" : { 
      "begin" : 125 ,
      "end" : 153 ,
      "surfaceForm" : "could potentially be used in"
    }
     ,
    "arguments" : [ { 
      "argnum" : 1 ,
      "begin" : 112 ,
      "end" : 124 ,
      "eid" : "E1" ,
      "canonicalForm" : "CI pipeline"
    }
    , { 
      "argnum" : 2 ,
      "begin" : 156 ,
      "end" : 180 ,
      "eid" : "E2" ,
      "canonicalForm" : "denial of service attack"
    }
    ,  ]
  }
  ,  ]
}
